src/ui/Base/ComponentOptions.ts

```diff
@@ -702,6 +707,16 @@ export class ComponentOptions {
     return Utils.isNonEmptyString(templateId) ? TemplateCache.getTemplate(templateId) : null;
   }
 
+  static loadResultTemplateFromPageTemplatesCache(): Template {
+    if (TemplateCache) {
+      var pageTemplateNames = TemplateCache.getPageTemplateNames();
+      if (pageTemplateNames.length > 0) {
+        return new TemplateList(_.compact(_.map(pageTemplateNames, (templateName) => TemplateCache.getTemplate(templateName))));
```

Comment:
Why does this need to be `_.compact` ? 

Is there any edge case where TemplateCache.getTemplate() would return null or undefined ? 

I think `.getTemplate` would throw anyway.